### DISCUSSION TITLE
BigQuery plugin

### DISCUSSION THREAD

**Discussion by simoncook** [Budibase/budibase#10355] [Help] [answered]
Being very noobie here, so please forgive if this is a stupid question. I'm completely baffled by how to add the BigQuery plugin from here https://github.com/marblekirby/budibase-big-query-plugin - could someone take pity on me and walk me through?

I got as far as downloading from https://github.com/marblekirby/budibase-big-query-plugin/releases/download/v1.0.0/big-query-1.0.0.tar.gz and then uploading this file but it says "Failed to add plugin: Failed to import plugin: Only component plugins are supported outside of self-host"

**Comment by andz-bb** [comment, marked as answer]
hey @simoncook, unfortunately our cloud environment doesn't allow you to add custom datasource plugins for security reasons. Sorry for the inconvenience!

If you'd like to use the BigQuery plugin, I'd suggest setting up a self-hosted Budibase environment if that's suitable for your use case. Our docs site has comprehensive guides for setting this up here: https://docs.budibase.com/docs/hosting-methods

Hope this helps!
